src/devices/bus/multibus/robotron_k7070.h

```diff
@@ -0,0 +1,76 @@
+// license:BSD-3-Clause
+// copyright-holders:Sergey Svishchev
+
+#ifndef MAME_BUS_MULTIBUS_ROBOTRON_K7070_H
+#define MAME_BUS_MULTIBUS_ROBOTRON_K7070_H
+
+#pragma once
+
+#include "multibus.h"
+
+#include "bus/rs232/rs232.h"
+#include "cpu/z80/z80.h"
+#include "machine/z80ctc.h"
+#include "machine/z80sio.h"
+
+#include "emupal.h"
+#include "screen.h"
+
+
+class robotron_k7070_device
+	: public device_t
+	, public device_multibus_interface
+{
+public:
+	robotron_k7070_device(machine_config const &mconfig, char const *tag, device_t *owner, u32 clock);
+
+protected:
+	virtual void device_add_mconfig(machine_config &config) override;
+	virtual void device_start() override;
+	virtual void device_reset() override;
+	virtual ioport_constructor device_input_ports() const override;
+	virtual const tiny_rom_entry *device_rom_region() const override;
+
+	void cpu_mem(address_map &map);
+	virtual void cpu_pio(address_map &map);
+
+private:
+	required_device<z80_device> m_cpu;
+	required_device<z80ctc_device> m_ctc;
+	required_device<z80sio_device> m_sio;
+	required_device_array<rs232_port_device, 2> m_serial;
+	required_device<palette_device> m_palette;
+	required_shared_ptr<u8> m_ram;
+
+	memory_view m_view_lo;
+	memory_view m_view_hi;
+
+	u16 m_start;
+
+	uint8_t kgs_host_r(offs_t offset);
+	void kgs_host_w(offs_t offset, uint8_t data);
+	void abg_addr_w(offs_t offset, uint8_t data);
+	void abg_func_w(offs_t offset, uint8_t data);
+	void abg_split_w(offs_t offset, uint8_t data);
+	void abg_misc_w(offs_t offset, uint8_t data);
+	void kgs_memory_remap();
+
+	bool m_abg_msel = 0, m_kgs_iml = 0;
+	uint8_t m_abg_func = 0, m_abg_split = 0;
+	uint16_t m_abg_addr = 0;
+	uint8_t m_kgs_datao = 0, m_kgs_datai = 0, m_kgs_ctrl = 0;
+	bool m_nmi_enable = 0;
```

Comment:
Please initialise members in the constructor initialiser list.  It’s confusing when there are member initialisations split between the header and source file.